ERCS/eip-xxxx.md

```diff
@@ -0,0 +1,266 @@
+
+However, due to the presence of wildcards there is additional work that needs to be done as opposed to a simple lookup.
+
+For example the method `eth_sendTransaction` there can be wildcards with the `ToAddress`, `Value` , `ChainId` and `FunctionSelector` . Thus, we need to try to match all permutations with the wildcard substituted. In the case of `eth_sendTransaction`, that's 2^4 = 16 lookups. There are more efficient datastructures that can used here but will be omitted for brevity. 
+
+Since this is an allowlist only, there is no precedence we need to take into account. If any one of the 16 permutations is found, the transaction is allowed.
+
+Here's a simplified diagram of the trie.
+
+![[../assets/eip-xxxx/DPF_eth_sendTransaction_trie.png]]
+
+If at any point a node is not found, this transaction is not allowed in the dApps Permission Framework.
+
+> [!todo]
+> Need to add limits to discuss DoS attacks
+> Performance.
+
+#### `Interaction` Merkle Patricia Tree Storage
+
+The Merkle Patricia Tree can be stored anywhere since we will verify the root hash in the DNS.
+
+The simplest example for storage is an open source Github repository containing the domain, root hash and nodes.
+
+#### DNS Compromise
+
+The entire previous specification we have assumed the DNS records are not tampered with. We will now relax that constraint and add protection from DNS compromises.
+
+We will introduce a 72 hour time lock whereby the old DNS records are respected. This gives teams enough time to detect compromises and regain control of their records or alert users.
+
+Let's walk through what happens when the DNS record is updated.
+
+If the DNS record is updated, wallets are expected to respect the previous dApp Permission Framework policy. Wallets can detect a DNS record change by referencing the root hash with the latest root hash stored for this domain.
+
+The simplest implementation would be to trust the open source Github repository. The open source Github repository will always contain the latest valid dApp Permission Framework for each domain. That is, the team who is updating their DNS records is required to create a pull request into the Github repository. However, this will not get accepted until the root hash in the pull request matches the domain and 72 hours has passed.
+	
+An attacker compromising the DNS record would also have to compromise the open source Github repository simultaneously in order to trick wallets. Thus, at all times there are two simultaneous attacks needed. Wallets caring about an extra layer of security can maintain mirrors this Github repository internally to ensure it hasn't been tampered with.
+
+Note: this means, teams should update their DNS Records to add more permissions **before** updating the dApp itself.
+
+### How it all fits together
+
+Let's walk through the full example for `opensea.io`.
+
+#### Developers
+
+> [!todo]
+> Fill this with exact values.
```

Comment:
If you use HTML-style comments (`<!-- TODO -->`), the linter will make sure you replace them before going into review.